pkg/assetfetcher/http.go

```diff
@@ -131,50 +103,81 @@ func (c *NexposeAssetFetcher) FetchAssets(ctx context.Context, siteID string, sc
 	return pagedAssetChan, pagedErrChan
 }
 
-func (c *NexposeAssetFetcher) makeRequest(ctx context.Context, wg *sync.WaitGroup, siteID string, scanID string, page int, assetChan chan domain.AssetEvent, errChan chan error) {
-	defer wg.Done()
+// fetchAllNexposeSiteAssets returns all assets from a given Nexpose site. This function first makes a call
+// to retrieve the first page of assets in a Nexpose site, then concurrently retrieves all pages after knowing the total pages.
+// The function uses a fanout pattern with no partial error
+func (c *NexposeAssetFetcher) fetchAllNexposeSiteAssets(ctx context.Context, siteID string) ([]Asset, error) {
+	// have to page through to get all the assets, start with page 0
+	// make the first call to Nexpose to get the total number of pages we'll
+	var currentPage = 0
+	ctx, cancel := context.WithCancel(ctx)
+	firstPageOfAssets, initialErr := c.fetchNexposeSiteAssetsPage(ctx, currentPage, siteID)
+	if initialErr != nil {
+		defer cancel()
```

Comment:
Task: This `cancel` is unnecessary. Please remove it.

Reply:
mmm i have some questions that need to be clarified, ill ask tomorrow